Tests/XcodeGenKitTests/SourceGeneratorTests.swift

```diff
@@ -390,6 +390,38 @@ class SourceGeneratorTests: XCTestCase {
                 try pbxProj.expectFile(paths: ["../OtherDirectory/C/D", "e.swift"], names: ["D", "e.swift"], buildPhase: .sources)
                 try pbxProj.expectFile(paths: ["Sources/B", "b.swift"], names: ["B", "b.swift"], buildPhase: .sources)
             }
+            
+            $0.it("generates custom groups") {
+                
+                let directories = """
+                Sources:
+                  A:
+                    - b.swift
+                  F:
+                    - G:
+                      - h.swift
+                  B:
+                    - b.swift
+                    - C:
+                      - c.swift
+                """
+                try createDirectories(directories)
+                
+                let target = Target(name: "Test", type: .application, platform: .iOS, sources: [
+                    TargetSource(path: "Sources/A/b.swift", group: "CustomGroup1"),
+                    TargetSource(path: "Sources/F/G/h.swift", group: "CustomGroup1"),
```

Comment:
What happens if there are multiple targetSource directories that each contain files in them, and share the same custom group. Do you they merged? eg including `Sources/G` and `Sources/H` that each have multiple files in them

Reply:
It should, I can add another test for this